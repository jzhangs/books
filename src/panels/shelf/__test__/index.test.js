import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Shelf from '..';

configure({ adapter: new Adapter() });

describe('Shelf', function() {
  it('renders correctly', () => {
    const wrapper = mount(<Shelf/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('show list when book.length not 0', () => {
    const wrapper = mount(<Shelf/>);
    wrapper.setState({ books: [ {'tile' : 'book1'} ]});
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should respond to keydown', () => {
    window.MozActivity = function () {};
    const spy = jest.spyOn(Shelf.prototype, 'onKeyDown');
    const wrapper = mount(<Shelf/>);
    wrapper.simulate('keydown', { key: 'SoftLeft' });
    wrapper.simulate('keydown', { key: '5' });
    expect(spy).toHaveBeenCalledTimes(2);
  });
});
