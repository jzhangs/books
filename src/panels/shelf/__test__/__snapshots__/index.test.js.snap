// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shelf renders correctly 1`] = `
<Shelf>
  <div
    className="shelf"
    onKeyDown={[Function]}
    role="menu"
    tabIndex="-1"
  >
    <Header
      text="books"
    >
      <header
        className="kai-header"
      >
        <h1
          className="h1"
          data-l10n-id="books"
        />
      </header>
    </Header>
    <div
      className="shelf-content"
    >
      <Empty
        text="no-books"
      >
        <div
          className="kai-empty"
        >
          <p
            className="kai-empty-text"
            data-l10n-id="no-books"
          />
        </div>
      </Empty>
    </div>
    <SoftKey
      left="open"
    >
      <form
        className="kai-softkey visible"
        data-type="action"
      >
        <Button
          content="open"
          pos="left"
        >
          <button
            className="kai-softkey-btn"
            data-l10n-id="open"
          />
        </Button>
        <Button
          content=""
          pos="center"
        >
          <button
            className="kai-softkey-btn"
          />
        </Button>
        <Button
          content=""
          pos="right"
        >
          <button
            className="kai-softkey-btn"
          />
        </Button>
      </form>
    </SoftKey>
  </div>
</Shelf>
`;

exports[`Shelf show list when book.length not 0 1`] = `
<Shelf>
  <div
    className="shelf"
    onKeyDown={[Function]}
    role="menu"
    tabIndex="-1"
  >
    <Header
      text="books"
    >
      <header
        className="kai-header"
      >
        <h1
          className="h1"
          data-l10n-id="books"
        />
      </header>
    </Header>
    <div
      className="shelf-content"
    />
    <SoftKey
      left="open"
    >
      <form
        className="kai-softkey visible"
        data-type="action"
      >
        <Button
          content="open"
          pos="left"
        >
          <button
            className="kai-softkey-btn"
            data-l10n-id="open"
          />
        </Button>
        <Button
          content=""
          pos="center"
        >
          <button
            className="kai-softkey-btn"
          />
        </Button>
        <Button
          content=""
          pos="right"
        >
          <button
            className="kai-softkey-btn"
          />
        </Button>
      </form>
    </SoftKey>
  </div>
</Shelf>
`;
